src/reconnectcontroller/DefaultReconnectController.ts

```diff
@@ -25,24 +30,47 @@ export default class DefaultReconnectController implements ReconnectController {
     return Date.now() - this.firstConnectionAttemptTimestamp;
   }
 
+  private timeSinceLastActiveMs(): number {
```

Comment:
Rather than implement all of this logic here and making the `ReconnectController` more complicated, I would suggest making a new monitor, e.g. `idlemonitor/IdleMonitor.ts` to handle the *detection* of the idle state. Then `ReconnectController` could then ask `IdleMonitor.isIdle()` in `hasPastReconnectDeadline` to determine whether it is allowed to reconnect.